lib/extra_opts.c

```diff
@@ -93,14 +93,14 @@ char **np_extra_opts(int *argc, char **argv, const char *plugin_name){
 			/* append the list to extra_args */
 			if(extra_args==NULL){
 				extra_args=ea1;
-				while(ea1=ea1->next) ea_num++;
+				while((ea1 = ea1->next)) ea_num++;
 			}else{
 				ea_tmp=extra_args;
 				while(ea_tmp->next) {
 					ea_tmp=ea_tmp->next;
 				}
 				ea_tmp->next=ea1;
-				while(ea1=ea1->next) ea_num++;
```

Comment:
what's the reason for adding duplicate brackets? And why change line 96 and 103 into different styles?

Reply:
```
warning: suggest parentheses around assignment used as truth value [-Wparentheses]
```
Arguably this one is not so important.

As for changing the style, I think it is easier to read with spaces around `=`, I don't know why I skipped it for 103.